File: app/src/main/java/com/cashzhang/nozdormu/FeedItem.java
package com.cashzhang.nozdormu;

/**
 * Created by cz21 on 2018/2/8.
 */

import java.io.IOException;
import java.io.ObjectInputStream;
import java.io.ObjectOutputStream;
import java.io.Serializable;

public class FeedItem implements Serializable {
    private static final long serialVersionUID = 3L;

    public String m_webtitle = "";
    public String m_title = "";
    public String m_imageLink = "";
    public String m_imageName = "";
    public String m_urlTrimmed = "";
    public String m_url = "";
    public String m_content = "";
    public String m_tcontent = "";
    public String[] m_desLines = {"", "", ""};
    public Long m_time = 0L;

    private void writeObject(ObjectOutputStream out) throws IOException {
        out.writeUTF(m_webtitle);
        out.writeUTF(m_title);
        out.writeUTF(m_imageLink);
        out.writeUTF(m_imageName);
        out.writeUTF(m_urlTrimmed);
        out.writeUTF(m_url);
        out.writeUTF(m_content);
        out.writeUTF(m_tcontent);
        out.writeObject(m_desLines);
        out.writeLong(m_time);
    }

    private void readObject(ObjectInputStream in) throws IOException, ClassNotFoundException {
        m_webtitle = in.readUTF();
        m_title = in.readUTF();
        m_imageLink = in.readUTF();
        m_imageName = in.readUTF();
        m_urlTrimmed = in.readUTF();
        m_url = in.readUTF();
        m_content = in.readUTF();
        m_tcontent = in.readUTF();
        m_desLines = (String[]) in.readObject();
        m_time = in.readLong();
    }

}

File: app/src/main/java/com/cashzhang/nozdormu/FeedItem.kt
package com.cashzhang.nozdormu

import java.io.IOException
import java.io.ObjectInputStream
import java.io.ObjectOutputStream
import java.io.Serializable

/**
 * Created by cz21 on 2018/2/8.
 */
class FeedItem : Serializable {
    private var webtitle = ""
    private var title = ""
    private var imageLink = ""
    private var imageName = ""
    private var urlTrimmed = ""
    private var url = ""
    private var content = ""
    private var tcontent = ""
    private var desLines = arrayOf("", "", "")
    private var time = 0L

    @Throws(IOException::class)
    private fun writeObject(out: ObjectOutputStream) {
        out.writeUTF(webtitle)
        out.writeUTF(title)
        out.writeUTF(imageLink)
        out.writeUTF(imageName)
        out.writeUTF(urlTrimmed)
        out.writeUTF(url)
        out.writeUTF(content)
        out.writeUTF(tcontent)
        out.writeObject(desLines)
        out.writeLong(time)
    }

    @Throws(IOException::class, ClassNotFoundException::class)
    private fun readObject(`in`: ObjectInputStream) {
        webtitle = `in`.readUTF()
        title = `in`.readUTF()
        imageLink = `in`.readUTF()
        imageName = `in`.readUTF()
        urlTrimmed = `in`.readUTF()
        url = `in`.readUTF()
        content = `in`.readUTF()
        tcontent = `in`.readUTF()
        desLines = `in`.readObject() as Array<String>
        time = `in`.readLong()
    }

    companion object {
        private const val serialVersionUID = 3L
    }
}